airbyte-integrations/connectors/source-hubspot/source_hubspot/source.py

```diff
@@ -2,25 +2,202 @@
+                name="engagements_meetings",
+                **common_params,
+            ),
+            CRMSearchStream(
+                entity="notes",
+                last_modified_field="hs_lastmodifieddate",
+                associations=["contacts", "deal", "company"],
+                name="engagements_notes",
+                **common_params,
+            ),
+            CRMSearchStream(
+                entity="tasks",
+                last_modified_field="hs_lastmodifieddate",
+                associations=["contacts", "deal", "company"],
+                name="engagements_tasks",
+                **common_params,
+            ),
+            CRMObjectIncrementalStream(
+                entity="feedback_submissions", associations=["contacts"], name="feedback_submissions", **common_params
+            ),
+            Forms(**common_params),
+            FormSubmissions(**common_params),
+            CRMObjectIncrementalStream(entity="line_item", name="line_items", **common_params),
+            MarketingEmails(**common_params),
+            Owners(**common_params),
+            CRMObjectIncrementalStream(entity="product", name="products", **common_params),
+            PropertyHistory(**common_params),
+            SubscriptionChanges(**common_params),
+            CRMObjectIncrementalStream(entity="ticket", name="tickets", **common_params),
+            TicketPipelines(**common_params),
+            Workflows(**common_params),
+        ]
+
+        credentials_title = credentials.get("credentials_title")
+        if credentials_title == "API Key Credentials":
+            streams.append(CRMObjectIncrementalStream(entity="quote", name="quotes", **common_params))
+
+        return streams
+
+    def read(
+        self, logger: logging.Logger, config: Mapping[str, Any], catalog: ConfiguredAirbyteCatalog, state: MutableMapping[str, Any] = None
+    ) -> Iterator[AirbyteMessage]:
+        """
+        This method is overridden to check whether the stream `quotes` exists in the source, if not skip reading that stream.
+        """
+        connector_state = copy.deepcopy(state or {})
+        logger.info(f"Starting syncing {self.name}")
+        config, internal_config = split_config(config)
+        # TODO assert all streams exist in the connector
+        # get the streams once in case the connector needs to make any queries to generate them
+        stream_instances = {s.name: s for s in self.streams(config)}
+        self._stream_to_instance_map = stream_instances
+        with create_timer(self.name) as timer:
+            for configured_stream in catalog.streams:
+                stream_instance = stream_instances.get(configured_stream.stream.name)
+                if not stream_instance and configured_stream.stream.name == "quotes":
+                    logger.warning("Stream `quotes` does not exist in the source. Skip reading `quotes` stream.")
+                    continue
```

Comment:
This method is overridden to check whether the stream `quotes` exists in the source, if not skip reading that stream.